src/app/app.component.ts

```diff
@@ -21,4 +26,37 @@ export class AppComponent {
       return 'off-home';
     }
   }
+
+  ngDoCheck() {
+    if (this.auth.authenticated() && this.currentUserId == null)
+    {
+      //console.log('on do check');
+      this.currentUserId = this.auth.getCurrentUserId();
+      //console.log("user id: " + this.currentUserId);
```

Comment:
i see a lot of `console.log()`, with Chrome devtools you can use a full powered debugger (with breakpoints in Typescript sourcemaps). at the very least, debug code like `console.log` should not be committed to the repo.